### PR TITLE
fix(suggest-next): add debug-session awareness to fix) case

### LINKED ISSUE #411: fix(suggest-next): fix) case doesn't check for active debug sessions
## Problem

`scripts/suggest-next.sh` `fix)` case unconditionally suggests `/vbw:qa -- Verify the fix`. If a user runs `/vbw:fix` while a debug session is active (e.g., in `qa_pending` or `qa_failed` state), the suggestion points to standalone QA instead of `/vbw:debug --resume`.

## Root cause

The `fix)` case has no debug-session-aware branch. It was written before debug sessions existed and was never updated.

## Affected components

- `scripts/suggest-next.sh` — `fix)` case (around line 720)

## Acceptance criteria

1. When a debug session is active and `fix)` is the command context, suggest `/vbw:debug --resume` instead of `/vbw:qa`
2. When no debug session is active, the existing behavior is preserved (suggest `/vbw:qa`)
3. BATS test coverage for the new branch

## Scope boundary

This is a standalone fix for pre-existing behavior. Not related to #405 (which addresses the debug/qa/verify flow).

### PR DESCRIPTION
**Pull request by dpearson2699**
## Linked Issue

Fixes #411

## What

Adds debug-session awareness to the `fix)` case in `scripts/suggest-next.sh`. When a debug session is active (standalone, no phases), the fix now suggests `/vbw:debug --resume` instead of `/vbw:qa` for most session states.

## Why

The `fix)` case was written before debug sessions existed and never updated. When a user ran `/vbw:fix` during an active debug session, they were directed to standalone QA (`/vbw:qa`) instead of resuming the debug session — losing debug-session context.

## How

- **`scripts/suggest-next.sh`**: Added a debug-session detection block to the `fix)` case (lines 717-754) using the identical pattern from `qa)` and `verify)` cases — `debug-session-state.sh get-or-latest` → `eval` → branch on `status`. Routes `qa_pending|qa_failed|fix_applied` → `/vbw:debug --resume -- Address remaining issues`, `uat_failed` → `/vbw:debug --resume -- Address UAT issues`, `investigating` → `/vbw:debug --resume -- Continue debugging`, `uat_pending` → `/vbw:qa -- Verify the fix`. The `phase_count=0` guard matches the convention in `qa)` and `verify)`.

- **`tests/suggest-next-debug-session.bats`**: Added 9 new BATS tests covering all debug session statuses (`qa_pending`, `qa_failed`, `fix_applied`, `uat_pending`, `uat_failed`, `investigating`), plus boundary cases (no session, complete session, phases-exist deferral).

## Acceptance Criteria Verification

1. **When a debug session is active and `fix)` is the command context, suggest `/vbw:debug --resume` instead of `/vbw:qa`**: Satisfied by `scripts/suggest-next.sh` lines 728-746, tested in `tests/suggest-next-debug-session.bats` tests 'fix with qa_pending/qa_failed/fix_applied/uat_failed/investigating debug session suggests resume'
2. **When no debug session is active, the existing behavior is preserved (suggest `/vbw:qa`)**: Satisfied by `scripts/suggest-next.sh` lines 749-752, tested in 'fix with no debug session suggests qa' and 'fix with complete debug session suggests qa'
3. **BATS test coverage for the new branch**: 9 new tests in `tests/suggest-next-debug-session.bats` covering every status branch plus edge cases

## Testing

- [x] Loaded plugin locally with `claude --plugin-dir .`
- [x] All 2763 BATS tests pass (`bash testing/run-all.sh`)
- [x] All 38 contract checks pass
- [x] All lint checks pass (ShellCheck, bash -n)
- [x] 9 new tests cover every code path in the new `fix)` debug-session block

## QA Summary

- **Primary QA (Claude)**: 3 rounds, zero findings across all rounds
- **Cross-model QA (GPT-5.4)**: 1 round, 1 medium finding (F-01: `phase_count=0` guard limits fix to standalone projects) — classified as false positive (intentional convention matching `qa)` and `verify)` cases; issue scope explicitly says 'standalone fix')
- **Copilot PR review**: Pending

### CODE REVIEW

File: scripts/suggest-next.sh [hunk 1]
```diff
@@ -715,8 +715,42 @@ case "$CMD" in
     ;;
 
   fix)
-    suggest "/vbw:qa -- Verify the fix"
-    suggest "/vbw:vibe -- Continue building"
+    # Check for active debug session (phase_count=0 only —
+    # when phases exist, fix.md itself handles debug-session next-step text
+    # inline, so we must not hijack phase fix suggestions here)
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/suggest-next.sh]: Fixed in a6b7dcf. Adopted the suggested wording — the comment now accurately describes the `phase_count=0` guard without claiming `fix.md` handles inline debug-session routing.

File: scripts/suggest-next.sh [hunk 2]
```diff
@@ -715,8 +715,41 @@ case "$CMD" in
...
+      if [ -n "$_fix_ds_output" ]; then
+        eval "$_fix_ds_output" 2>/dev/null || true
+        # shellcheck disable=SC2154
+        if [ "${active_session:-none}" != "none" ] && [ -n "${session_file:-}" ] && [ -f "$session_file" ]; then
+          _fix_ds_status="${status:-}"
+          case "$_fix_ds_status" in
+            qa_pending|qa_failed|fix_applied)
+              suggest "/vbw:debug --resume -- Address remaining issues"
+              _fix_debug_handled=true
+              ;;
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/suggest-next.sh]: Fixed in ee45652. Split `qa_failed` into its own branch with "Address QA failures" message for consistency with the `qa)` and `debug)` cases. Updated the corresponding BATS test assertion.

File: scripts/suggest-next.sh [hunk 3]
```diff
@@ -715,8 +715,45 @@ case "$CMD" in
...
+            uat_pending)
+              suggest "/vbw:qa -- Verify the fix"
+              _fix_debug_handled=true
+              ;;
+            uat_failed)
+              suggest "/vbw:debug --resume -- Address UAT issues"
+              _fix_debug_handled=true
+              ;;
+            investigating)
+              suggest "/vbw:debug --resume -- Continue debugging"
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/suggest-next.sh]: Fixed in c26170f. Changed "Continue debugging" to "Continue investigation" to match the `debug)` case. Updated the BATS test assertion accordingly.

File: scripts/suggest-next.sh [hunk 4]
```diff
@@ -715,8 +715,45 @@ case "$CMD" in
...
+            qa_pending|fix_applied)
+              suggest "/vbw:debug --resume -- Address remaining issues"
+              _fix_debug_handled=true
+              ;;
+            qa_failed)
+              suggest "/vbw:debug --resume -- Address QA failures"
+              _fix_debug_handled=true
+              ;;
+            uat_pending)
+              suggest "/vbw:qa -- Verify the fix"
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply scripts/suggest-next.sh]: Fixed in 758981d. Changed `uat_pending` to suggest `/vbw:verify -- Run UAT on the debug fix` instead of `/vbw:qa`, matching the debug-session lifecycle where `uat_pending` means UAT should be run next.

File: tests/suggest-next-debug-session.bats [hunk 5]
```diff
@@ -163,3 +163,77 @@ EOF
...
+  run bash "$SCRIPTS_DIR/suggest-next.sh" fix pass
+  [ "$status" -eq 0 ]
+  [[ "$output" == *"--resume"* ]]
+}
+
+@test "suggest-next fix with uat_pending debug session suggests qa" {
+  create_debug_session "uat_pending"
+  run bash "$SCRIPTS_DIR/suggest-next.sh" fix pass
+  [ "$status" -eq 0 ]
+  [[ "$output" == *"/vbw:qa"* ]]
```
[the remark this answers is not in the document]
Comment (dpearson2699) [reply tests/suggest-next-debug-session.bats]: Fixed in 758981d. Updated the test to assert `/vbw:verify` and `UAT` for the `uat_pending` case.
